typescript/sdk/src/token/adapters/SealevelTokenAdapter.ts

```diff
@@ -108,11 +108,18 @@ export class SealevelTokenAdapter
   }
 
   async getBalance(owner: Address): Promise<bigint> {
-    const tokenPubKey = this.deriveAssociatedTokenAccount(new PublicKey(owner));
-    const response = await this.getProvider().getTokenAccountBalance(
-      tokenPubKey,
-    );
-    return BigInt(response.value.amount);
+    try {
+      const tokenPubKey = this.deriveAssociatedTokenAccount(
+        new PublicKey(owner),
+      );
```

Comment:
should we keep this derivation outside of the try/catch?

Reply:
Good idea, done.